ddtrace/contrib/asyncio/helpers.py

```diff
@@ -72,3 +73,54 @@ def _wrap_executor(fn, args, tracer, ctx):
     # fn() will be executed outside the asyncio loop as a synchronous code
     tracer._context_provider._local.set(ctx)
     return fn(*args)
+
+
+_orig_create_task = None
+
+
+def enable_task_linking():
```

Comment:
In that case we're providing a generic enabled / disabled flag that monkey patches the stdlib (for obvious reasons because probably there aren't so many different ways to enable that by default). While it's interesting to provide an automatic full ON / full OFF functionality, I think that it could be better and a bit more generic to provide a helper function that can be used to say: "let's propagate the Context only for this Task".

This means that the underlying `_create_task` may become a helper function like `run_in_executor` ([ref][1]) so that it's explicit when you're going to propagate the context to a new task:
```python
from ddtrace.contrib.asyncio import helpers

# propagates the context
helpers.create_task(coro())

# doesn't propagate the context
loop.create_task(coro())
```

The main reason is that in some cases we don't want the automatic propagation when creating new tasks (e.g. a very long-running Task that lasts 2+ minutes could break the stats for users requests). On the other hand, that helper should be enough to wrap easily the `BaseEventLoop.create_task`, if someone needs to enable that by default everywhere.

Is it reasonable for you if we move that function as a helper that must be used manually, so that you can later wrap your `loop.create_task` with this one in your code base?
If it is, we may think that your `_create_task` is the first building block that anyone can use to manually or automatically propagate the `Context`. Later on, we may think to keep it enabled by default but for the moment, maybe it's more generic to provide a helper and some patterns to enable it.

[1]: https://github.com/thehesiod/dd-trace-py/blob/b9a41ec6cd7f8636cdbddb2134eaab88604f085e/ddtrace/contrib/asyncio/helpers.py#L36